apps/src/bin/quiche-server.rs

```diff
@@ -65,37 +68,61 @@ fn main() {
     let conn_args = CommonArgs::with_docopt(&docopt);
     let args = ServerArgs::with_docopt(&docopt);
 
+    let preferred_address: SocketAddr = args
+        .preferred_address
+        .parse()
+        .expect("Unable to parse preferred address.");
+
     // Setup the event loop.
     let mut poll = mio::Poll::new().unwrap();
     let mut events = mio::Events::with_capacity(1024);
 
     // Create the UDP listening socket, and register it with the event loop.
-    let mut socket =
+    let mut starting_socket =
         mio::net::UdpSocket::bind(args.listen.parse().unwrap()).unwrap();
 
-    // Set SO_TXTIME socket option on the listening UDP socket for pacing
-    // outgoing packets.
-    if !args.disable_pacing {
-        match set_txtime_sockopt(&socket) {
-            Ok(_) => {
-                pacing = true;
-                debug!("successfully set SO_TXTIME socket option");
-            },
-            Err(e) => debug!("setsockopt failed {:?}", e),
-        };
+    // Create another preferred UDP listening socket that can be used for server
+    // migration.
+    let mut preferred_socket =
+        mio::net::UdpSocket::bind(preferred_address).unwrap();
```

Comment:
Can this preferred socket take new connections not just migrated connections?
If a new connection starts here, is the preferred address provided over the transport params? What does this cause? - Lucas

Reply:
Error Result if preferred address is same as current address - fail on startup - Lucas

Reply:
Yeah that is a great question.
For the example app the app will accept connections starting on the servers preferred address. The server will still encode a "preferred address" transport parameter into the transport parameters it initially sends to the client. The client will receive the transport parameters and begin decoding them. It will see an encoded preferred address transport parameter and begin decoding it. The example app will then initiate `probe_path()` on this "new" preferred address. The `probe_path()` logic sees that this is an existing path. It will generate a Path_Challenge frame and then it does not do path migration since there in no migration to perform.

This separation of concerns makes sense to me. The probe_path / connection migration code should be resilient enough to not attempt migrations to the path it is already on.

<img width="1512" alt="Screenshot 2024-08-01 at 2 07 11 PM" src="https://github.com/user-attachments/assets/1fe5f4e5-3f32-4cfa-9e33-0fb5b8a2ea57">

Reply:
Lucas: "Add a comment to explicitly state that this is intentional and not a bug"
Lucas to follow up with working group to further discuss.